torch_xla/experimental/pallas_kernels/ragged_paged_attention_v2.py

```diff
@@ -1954,12 +1954,14 @@ def masked_store(ref, val, start, end, group=1):
       # kv lens will be contracting dim, we should mask out the NaNs.
       kv_mask = (
           lax.broadcasted_iota(jnp.int32, k.shape, 0) < kv_len - kv_len_start)
-      k = jnp.where(kv_mask, k.astype(jnp.float32), 0).astype(k.dtype)
-      v = jnp.where(kv_mask, v.astype(jnp.float32), 0).astype(v.dtype)
+      k = jnp.where(kv_mask, k, 0)
+      v = jnp.where(kv_mask, v, 0)
 
       qk = (
           jnp.einsum("nd,md->nm", q, k, preferred_element_type=jnp.float32) *
           sm_scale)
+      if k_scale is not None:
+        qk *= k_scale
```

Comment:
It will become w8a8 instead of w8a16, right?

Reply:
Thanks! I think we can multiply  k_scale and sm_scale first and then qk *= combined_scale. Thus the first one is a scalar multiplication.

Reply:
I believe the compiler already takes care of this, but it's never a bad idea to be safe. Changed the code like you've suggested.